rc/tools/spell.kak

```diff
@@ -142,20 +142,27 @@ define-command spell-next %{ evaluate-commands %sh{
 
 define-command \
     -docstring "Suggest replacement words for the current selection, against the last language used by the spell-check command" \
-    spell-replace %{ evaluate-commands %sh{
-    options=""
-    if [ -n "$kak_opt_spell_last_lang" ]; then
-        options="-l '$kak_opt_spell_last_lang'"
-    fi
-    suggestions=$(printf %s "$kak_selection" | eval "aspell -a $options" | grep '^&' | cut -d: -f2)
-    menu=$(printf %s "${suggestions#?}" | awk -F', ' '
-        {
-            for (i=1; i<=NF; i++)
-                printf "%s", "%{"$i"}" "%{execute-keys -itersel %{c"$i"<esc>be}}"
+    spell-replace %{
+    prompt \
+        -init %val{selection} \
+        -shell-script-candidates %{
+            options=""
+            if [ -n "$kak_opt_spell_last_lang" ]; then
+                options="-l '$kak_opt_spell_last_lang'"
+            fi
+            printf %s "$kak_selection" |
+                eval "aspell -a $options" |
+                kak -f '<a-s><a-K>^&<ret>d%s^[^:]*: <ret>d%s, <ret>c<ret><esc>'
```

Comment:
Could we not re-implement this with Awk? Otherwise, using `-n` as well would help loading time.

Reply:
Using `-n` wouldn't help; Kakoune checks for and executes `-f` long before it even bothers checking for `-n`.

I used `kak -f` because I couldn't immediately see how to perform the same operations with more primitive tools, but now that I've stepped away for a few hours I figured out how to do it with pure `sed`.